### ISSUE TITLE
meow--current-state cannot be used to tell meow state, when none of the state is active 

### ISSUE THREAD

**Issue by Inc0n** [meow-edit/meow#258]
`meow--current-state` cannot be used to tell meow state, when none of the state is active.

The usage arise from my custom modeline which relies on `meow--current-state` to tell meow state. When none of the state is active, `meow--current-state` will store the last active meow state, which isn't active.

A potential fix for this is to set `meow--current-state` to `nil`, in the way below:
```
(defun meow--define-state-minor-mode (name
                                      init-value
                                      description
                                      keymap
                                      lighter
                                      form)
  "Generate a minor mode definition with name meow-NAME-mode,
DESCRIPTION and LIGHTER."
  `(define-minor-mode ,(meow-intern name "-mode")
     ,description
     :init-value ,init-value
     :lighter ,lighter
     :keymap ,keymap
     (if (not ,(meow-intern name "-mode"))
         (setq-local meow--current-state nil)
       (meow--disable-current-state)
       (setq-local meow--current-state ',(intern name))
       (meow-update-display))
     ,form))
```

**Comment by DogLooksGood** [comment]
I need to review this. There are exit functions called when exiting a state, in `(meow--disable-current-state)`. I'm not sure whether we can set it to nil here. But there are predicate functions for state detection, e.g. `(meow-normal-mode-p)`.

**Comment by Inc0n** [comment]
Understandably so, I also see that meow--current-state is set to nil in `(meow--disable-current-state)`, which could mean a more appropriate solution to be calling it twice.

```
(if (not ,(meow-intern name "-mode"))
    (meow--disable-current-state)
  (meow--disable-current-state)
  (setq-local meow--current-state ',(intern name))
  (meow-update-display)
  ,form)
```

> predicate functions for state detection

As for predicate function for state detection, my usage involes detecting states flexibly across viper, evil and now meow also, using a state variable would be simpler to use (more desireble), and would be prefered personally.

**Comment by eshrh** [comment]
Pretty sure this is closed by your pr #259
